mta/ext.go

```diff
@@ -229,13 +229,17 @@ func extendMap(m *map[string]interface{}, meta map[string]MetaData, ext map[stri
 }
 
 func mergeMapKey(m *map[string]interface{}, key string, value interface{}) error {
-	extMapValue, extIsMap := getMapValue(value)
-	mMapValue, mIsMap := getMapValue((*m)[key])
+	extMapValue, extIsMap, _ := getMapValue(value)
+	mMapValue, mIsMap, converted := getMapValue((*m)[key])
 
 	if (*m)[key] == nil || value == nil || (!extIsMap && !mIsMap) {
 		(*m)[key] = value
 	} else if mIsMap && extIsMap {
-		if err := extendMap(&mMapValue, nil, extMapValue); err != nil {
+		err := extendMap(&mMapValue, nil, extMapValue)
+		if converted {
+			(*m)[key] = mMapValue
+		}
+		if err != nil {
```

Comment:
please move it up

Reply:
Move what up? The assignment should be done even in case of an error